src/app_service/service/community/service.nim

```diff
@@ -86,6 +86,62 @@ QtObject:
     result.allCommunities = initTable[string, CommunityDto]()
     result.myCommunityRequests = @[]
 
+  proc findChatById(id: string, chats: seq[ChatDto]): ChatDto =
+    for chat in chats:
+      if(chat.id == id):
+        return chat
+
+  proc findById(id: string, chats: seq[Chat]): int =
+    for chat in chats:
+      if(chat.id == id):
+        return 0
+    return -1
+
+  proc mapChatDtoToChat*(chat: ChatDto): Chat =
+    result = Chat()
+    result.id = chat.id
+    result.name = chat.name
+    result.color = chat.color
+    result.emoji = chat.emoji
+    result.description = chat.description
+    result.canPost = chat.canPost
+    result.position = chat.position
+
+  proc handleCommunityUpdates(self: Service, communities: seq[CommunityDto], updatedChats: seq[ChatDto]) =
+    let community = communities[0]
+    if(not self.joinedCommunities.hasKey(community.id)):
+      error "error: community doesn't exist"
+      return
+
+    let prev_community = self.joinedCommunities[community.id]
+
+    # channel was added
+    if(community.chats.len > prev_community.chats.len):
+      for chat in community.chats:
+        if findById(chat.id, prev_community.chats) == -1:
+          let updated_chat = findChatById(community.id&chat.id, updatedChats)
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_CREATED, CommunityChatArgs(communityId: community.id, chat: mapChatDtoToChat(updated_chat)))
+
+    # channel was removed
+    elif(community.chats.len < prev_community.chats.len):
+      for prv_chat in prev_community.chats:
+        if findById(prv_chat.id, community.chats) == -1:
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_DELETED, CommunityChatIdArgs(communityId: community.id, chatId: community.id&prv_chat.id))
+
+    # some property has changed
+    else:
+      for chat in community.chats:
+        # id is present
+        if findById(chat.id, prev_community.chats) == -1:
+          error "error: chat not present"
+          return
+        # but something is different
+        if(prev_community.chats.find(chat) == -1):
+          let updated_chat = findChatById(community.id&chat.id, updatedChats)
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_EDITED, CommunityChatArgs(communityId: community.id, chat: mapChatDtoToChat(updated_chat)))
```

Comment:
IMO, you can pass the chat as a ChatDto. ChatDto always has more data and it better than Community Chat.

Reply:
Hey @jrainville I discussed this with Sale and at first glance it is better too. However the ChatDto we receive in received Data and the 

> Code looks good for the most part.
> 
> I asked for changes, because I think it can be way simpler and also better if you pass the ChatDto as is instead of converting to Community Chat and then back to ChatDto.

I can't pass it as it is because the chatDto received doesn't contain information such as position etc, also sometimes what I have noticed is that the signal from status_go isn't sending all the chats which have been updated in the chatsDto and I need to access the receivedData.communities instead.

So, I can still pass a ChatDto by converting it from Community Chat to ChatDto and adding some missing params. maybe passing it as a ChatDto is more clean on the controller side.
wdyt?

Reply:
Ah ok I didn't know about the missing data.

The suggestion you put forth at the end seems reasonable. Do you want to do it in this PR or another?

Reply:
I am adding some code for position change. I can do it with that.
just wanted to add that there are also category updates which I haven't handled at all in this PR. I think its complex enough already, maybe in another GH issue for that.

Reply:
Ok perfect, let me know when you removed the leftover and fixed the function name then I can approve

Reply:
Done. And I did move the DTO part to the service side too :)